### PR TITLE
Access commands

### PR DESCRIPTION
**Pull request by danielwhatmuff**

### CODE REVIEW

File: app/views/integrations/_global_credentials.html.erb [hunk 1]
```diff
@@ -25,12 +25,52 @@
         class: 'collapse'
       ) do
     %>
+
       <dl class="mt-2 card p-3 bg-light overflow-auto">
-        <% credentials.each do |(k, v)| %>
-          <dt><%= k -%></dt>
-          <dd class="text-wrap text-break"><%= v -%></dd>
+        <%- if integration.provider_id == 'git_hub' %>
```
Comment (jits) [inline app/views/integrations/_global_credentials.html.erb]: Could you change these `if`s to a `case` statement ([example of where I did this in another PR](https://github.com/appvia/appvia-hub/pull/181/commits/69f8fec7862038e49be4211b29f40b5fc683f074#diff-8379ce8b652e259ffdbc3e809297b6e7R28)).

File: app/views/integrations/_global_credentials.html.erb [hunk 2]
```diff
@@ -25,12 +25,52 @@
         class: 'collapse'
       ) do
     %>
+
       <dl class="mt-2 card p-3 bg-light overflow-auto">
```
Comment (jits) [inline app/views/integrations/_global_credentials.html.erb]: `dl` tag should now be a `div`

**Comment by jits** [comment]
One thought: it may still be useful to show the "raw" credentials as before (since people may have their own ways to set things up), and then show the new instructions in a section below that, as a "and here's now how you can use these credentials to set robot access up".

File: app/views/integrations/_global_credentials.html.erb (hunk 1 above)
Comment (danielwhatmuff) [reply app/views/integrations/_global_credentials.html.erb]: sure

**Comment by danielwhatmuff** [comment]
> One thought: it may still be useful to show the "raw" credentials as before (since people may have their own ways to set things up), and then show the new instructions in a section below that, as a "and here's now how you can use these credentials to set robot access up".

Yeh that could work.

File: app/views/integrations/_global_credentials.html.erb (hunk 2 above)
Comment (danielwhatmuff) [reply app/views/integrations/_global_credentials.html.erb]: Done

File: app/views/integrations/_global_credentials.html.erb (hunk 1 above)
Comment (danielwhatmuff) [reply app/views/integrations/_global_credentials.html.erb]: Done

**Comment by jits** [comment]
>> One thought: it may still be useful to show the "raw" credentials as before (since people may have their own ways to set things up), and then show the new instructions in a section below that, as a "and here's now how you can use these credentials to set robot access up".
>
> Yeh that could work.

I don't think this PR should be merged until this is addressed.

**Comment by danielwhatmuff** [comment]
> > > One thought: it may still be useful to show the "raw" credentials as before (since people may have their own ways to set things up), and then show the new instructions in a section below that, as a "and here's now how you can use these credentials to set robot access up".
> > 
> > 
> > Yeh that could work.
> 
> I don't think this PR should be merged until this is addressed.

Ok, so shall I do a separate expandable box with the raw credentials (basically just add what was there before this PR below the commands box)?

**Comment by jits** [comment]
> Ok, so shall I do a separate expandable box with the raw credentials (basically just add what was there before this PR below the commands box)?

I'd suggest something like:

- First, show the "raw" credentials as-is (i.e. keep what we had before).
- Then below this, add your new expandable box with the commands to run (with some help text to explain that these are examples of how to take the credentials from above and set them up).

… this should give us enough to user test and iterate on. (Key thing being: how will user's react / what will they think when they see both sections?)

File: app/views/integrations/_global_credentials.html.erb [hunk 3]
```diff
@@ -1,36 +1,74 @@
 <% credentials = global_credentials_for integration %>
```
Comment (danielwhatmuff) [inline app/views/integrations/_global_credentials.html.erb]: This is how atom wants me to format html/erb :-S @jits

Comment (jits) [reply app/views/integrations/_global_credentials.html.erb]: I usually add the extra indentation myself.

Comment (danielwhatmuff) [reply app/views/integrations/_global_credentials.html.erb]: sorted..

Comment (danielwhatmuff) [reply app/views/integrations/_global_credentials.html.erb]: Ok should be good now.

File: app/views/integrations/_global_credentials.html.erb [hunk 4]
```diff
@@ -1,5 +1,38 @@
...
 <% if credentials.present? %>
+  <div class="<%= container_css_class if local_assigns.key? :container_css_class -%>">
+    <% panel_id = "credentials-panel-#{SecureRandom.hex(10)}" %>
```
Comment (jits) [inline app/views/integrations/_global_credentials.html.erb]: This is the same ID as the panel below… does it still work?

File: app/views/integrations/_global_credentials.html.erb [hunk 5]
```diff
@@ -1,5 +1,38 @@
...
+      aria: {
+        controls: panel_id
+      }
+    ) do
+    %>
+
+    <%= icon 'key'%>
+    Robot credentials
+    <%= icon 'caret-down', css_class: 'my-2' %>
+    <% end %>
```
Comment (jits) [inline app/views/integrations/_global_credentials.html.erb]: Indenting is still off here…

File: app/views/integrations/_global_credentials.html.erb [hunk 6]
```diff
@@ -1,5 +1,38 @@
...
+        class: 'collapse'
+      ) do
+    %>
+    <dl class="mt-2 card p-3 bg-light overflow-auto">
+      <% credentials.each do |(k, v)| %>
+      <dt><%= k -%></dt>
+      <dd class="text-wrap text-break"><%= v -%></dd>
+      <% end %>
+    </dl>
+    <% end %>
```
Comment (jits) [inline app/views/integrations/_global_credentials.html.erb]: … and here

File: app/views/integrations/_global_credentials.html.erb (hunk 4 above)
Comment (danielwhatmuff) [reply app/views/integrations/_global_credentials.html.erb]: Yeh it works. Guess that random string is generated separately for each instance. I will update the prefix though.

Comment (jits) [reply app/views/integrations/_global_credentials.html.erb]: Ah yes, that makes sense.

**Pull request merged**